### PR TITLE
Add DB::open_cf_with_ttl method.

### PR DESCRIPTION
**Pull request by fdeantoni**
Added the ability to open a database with column families and TTL. It uses method `ffi::rocksdb_open_column_families_with_ttl`.

### CODE REVIEW

**Review by aleksuss — COMMENTED** [review]
Overall looks good 👍🏻

File: src/db.rs [hunk 1]
```diff
@@ -110,6 +110,25 @@ impl DB {
         })
     }
 
+    /// Opens the database with a Time to Live compaction filter and column family names.
```
[the remark this answers is not in the document]
Comment (fdeantoni) [reply src/db.rs:113]: Thanks for reviewing this PR @aleksuss! I have pushed a new commit that contains the fixes you suggested. Let me know if this addresses what you were suggesting.

File: src/db.rs [hunk 2]
```diff
@@ -110,6 +110,42 @@ impl DB {
...
+        let cfs = cfs
+            .into_iter()
+            .map(|name| ColumnFamilyDescriptor::new(name.as_ref(), Options::default()));
+
+        DB::open_cf_descriptors_with_ttl(opts, path, cfs, ttl)
+    }
+
+    /// Opens a database with the given database with a Time to Live compaction filter and
+    /// column family descriptors.
+    pub fn open_cf_descriptors_with_ttl<P, I>(
```
Comment (aleksuss) [inline src/db.rs:136]: It seems this method is redundant.

File: src/db.rs [hunk 3]
```diff
@@ -110,6 +110,42 @@ impl DB {
...
+    where
+        P: AsRef<Path>,
+        I: IntoIterator<Item = N>,
+        N: AsRef<str>,
+    {
+        let cfs = cfs
+            .into_iter()
+            .map(|name| ColumnFamilyDescriptor::new(name.as_ref(), Options::default()));
+
+        DB::open_cf_descriptors_with_ttl(opts, path, cfs, ttl)
```
Comment (aleksuss) [inline src/db.rs:131]:
```suggestion
        DB::open_cf_descriptors_internal(opts, path, cfs, &AccessType::WithTTL { ttl })
```

File: src/db.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleksuss) [reply src/db.rs:113]: I mean to create a method like this:
```rust
pub fn open_cf_with_ttl_opt<P, I, N>(
         opts: &Options,
         path: P,
         cfs: I,
         ttl: Duration,
         cfs_opts: Options,
     ) -> Result<DB, Error>
     where
         P: AsRef<Path>,
         I: IntoIterator<Item = N>,
         N: AsRef<str>,
     {
         let cfs = cfs
             .into_iter()
             .map(|name| ColumnFamilyDescriptor::new(name.as_ref(), cfs_opts.clone()));

         DB::open_cf_descriptors_internal(opts, path, cfs, &AccessType::WithTTL { ttl })
}
```

**Comment by fdeantoni** [comment]
Thanks for the feedback @aleksuss. Just a question: 

I actually patterned `DB::open_cf_descriptors_with_ttl` after the existing `DB::open_cf_descriptors` method. I made `DB::open_cf_with_ttl()` use `DB::open_cf_descriptors_with_ttl()` so that the existing test would exercise both methods.

If `DB::open_cf_descriptors_with_ttl` is redundant, perhaps `DB::open_cf_descriptors` should be marked deprecated with a new method like the one you describe above (but without the TTL)? Something like:
```rust
    /// Opens a database with the given database options and column family descriptors.
    pub fn open_cf_opt<P, I>(opts: &Options, path: P, cfs: I, cfs_opts: Options) -> Result<DB, Error>
    where
        P: AsRef<Path>,
        I: IntoIterator<Item = ColumnFamilyDescriptor>,
    {
         let cfs = cfs
             .into_iter()
             .map(|name| ColumnFamilyDescriptor::new(name.as_ref(), cfs_opts.clone()));

        DB::open_cf_descriptors_internal(opts, path, cfs, &AccessType::ReadWrite)
    }
```

Follow up question: Implementing things this way does take away the ability to create / re-open a DB with ColumnFamilies having different configurations. I'm assuming there may be use cases where a database may want to have different configurations per column family?

**Comment by aleksuss** [comment]
> I'm assuming there may be use cases where a database may want to have different configurations per column family?

It is exactly what I've meant. But I rethought and I think the better way to do it in another PR.

**Review by aleksuss — APPROVED** [review]

**Pull request merged**